mailbox/api/src/main/java/org/apache/james/mailbox/indexer/MessageIdReIndexer.java

```diff
@@ -0,0 +1,27 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.indexer;
+
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.task.Task;
+
+public interface MessageIdReIndexer {
+    Task reIndex(MessageId messageId);
```

Comment:
why don't you place this method inside `ReIndexer` while creating a new interface for it?

Reply:
For the exact same reason described here https://github.com/linagora/james-project/pull/1808#discussion_r225387910